dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordMapper.cs

```
@@ -0,0 +1,99 @@
+    private readonly VectorStoreRecordPropertyReader _propertyReader;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqliteVectorStoreRecordMapper{TRecord}"/> class.
+    /// </summary>
+    /// <param name="propertyReader">A <see cref="VectorStoreRecordDefinition"/> that defines the schema of the data in the database.</param>
+    public SqliteVectorStoreRecordMapper(VectorStoreRecordPropertyReader propertyReader)
+    {
+        Verify.NotNull(propertyReader);
+
+        this._propertyReader = propertyReader;
+    }
+
+    public Dictionary<string, object?> MapFromDataToStorageModel(TRecord dataModel)
+    {
+        var properties = new Dictionary<string, object?>
+        {
+            // Add key property
+            { this._propertyReader.KeyPropertyStoragePropertyName, this._propertyReader.KeyPropertyInfo.GetValue(dataModel) }
+        };
+
+        // Add data properties
+        foreach (var property in this._propertyReader.DataPropertiesInfo)
+        {
+            properties.Add(this._propertyReader.StoragePropertyNamesMap[property.Name], property.GetValue(dataModel));
+        }
+
+        // Add vector properties
+        foreach (var property in this._propertyReader.VectorPropertiesInfo)
+        {
+            object? result = null;
+            var propertyValue = property.GetValue(dataModel);
+
+            if (propertyValue is not null)
+            {
+                var vector = (ReadOnlyMemory<float>)propertyValue;
+                result = SqliteVectorStoreRecordPropertyMapping.MapVector(vector);
+            }
+
+            properties.Add(this._propertyReader.StoragePropertyNamesMap[property.Name], result);
+        }
+
+        return properties;
+    }
+
+    public TRecord MapFromStorageToDataModel(Dictionary<string, object?> storageModel, StorageToDataModelMapperOptions options)
+    {
+        var record = (TRecord)this._propertyReader.ParameterLessConstructorInfo.Invoke(null);
```

You should call VerifyHasParameterlessConstructor in the constructor of this mapper to fail early if the data model doesn't have one.